ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/RequiredPluginsInitializer.java

```diff
@@ -85,4 +140,15 @@ public void requestClasspathContainerUpdate(IPath containerPath, IJavaProject pr
 		JavaCore.setClasspathContainer(containerPath, new IJavaProject[] {project}, new IClasspathContainer[] {containerSuggestion}, null);
 	}
 
+	private static boolean isCalledFromUI() {
+		try {
+			Class<?> display = RequiredPluginsInitializer.class.getClassLoader()
+					.loadClass("org.eclipse.swt.widgets.Display"); //$NON-NLS-1$
```

Comment:
If I see it right, this will trigger all static init code in the Display class to run if the display wasn't yet used (and also all the classes used by display static code), and that is potentially worse as the bug you are trying to workaround here.

Please don't do that.

Also please don't add (even if technically possible) dependency to SWT from PDE core. It is just not OK.

Reply:
I just replaced the check wit a stackwalker that detects if this is performed during the creation of the E4 Workbench (and therefore hinders the UI to show up).